src/foms.py

```diff
@@ -0,0 +1,186 @@
+        write(outfile,cli_args, f'     {ds_header}         {ds_header}         {ds_header}')
+        write(outfile,cli_args, )
+        for image_file_name in images:
+            matches = re.search(r'.*([0-9]{2})-([0-9]{2}).raw$', str(image_file_name))
+            iteration = int(matches.group(1))
+            subset    = int(matches.group(2))
+            foms = get_foms(commandStart + str(image_file_name))
+            crcs, snrs = foms
+            write(outfile,cli_args, f'{iteration:02}-{subset:02} ', end='')
+            crcs, variabilities = tuple(zip(*((crc, var) for (r, (crc, var)) in crcs.items())))
+            write(outfile,cli_args, ''.join(f'{c:6.1f} ' for c in crcs)         , end='         ')
+            write(outfile,cli_args, ''.join(f'{v:6.1f} ' for v in variabilities), end='         ')
+            write(outfile,cli_args, ''.join(f'{r:6.1f} ' for r in snrs.values())) # look broken in the Rust implementation of foms
+    
+def plot_from_fom(dir, sphere_diameters, cli_args):
+    count = 0
+    isData = False
+    data = dict()
+    for d in sphere_diameters:
+        data[d] = [] #initialize lists for each diameter key
+    with open(dir, encoding='utf-8') as f:
+        Lines = f.readlines()
+        for line in Lines:
+            if count == 4:
+                isData = True
+            if not isData:
+                count += 1
+                continue
+            #remove first element from a line and convert the remaining elements to floats
+            parsedLine = list(map(float, line.split()[1:]))
+            for i in range(len(sphere_diameters)):
+                data[sphere_diameters[i]].append((float(count-4), parsedLine[i],
+                                                  parsedLine[i+len(sphere_diameters)]))
+            count += 1
+    for d in sphere_diameters:
+        x = [data[d][i][0] for i in range(len(data[d]))]
+        y = [data[d][i][1] for i in range(len(data[d]))]
+        e = [data[d][i][2] for i in range(len(data[d]))]
+        plt.figure()
+        plt.plot(x,y,linewidth=2.0)
+        plt.errorbar(x,y,yerr=e,label=f'TOF=t c',capsize=3)
+        plt.legend()
+        plt.ylim(top=120)
+        plt.title(f'CRC vs iteration for {d}mm sphere')
+        plt.savefig(f'{cli_args["<DIR>"]}/crcs-{d}mm.png')
+
+    if cli_args['--show-plot']:
+        plt.show()
+
+def main():
+    cli_args = docopt(__doc__)
+    directory = cli_args['<DIR>']
+    print(f'Calculating FOMs for images in {directory}')
+    filename = f'{directory}/foms'
+    images = sorted(Path(directory).glob('*.raw'))
+
+    if bool(cli_args['<jaszczak>']):
+        commandStart = 'target/release/foms -- jaszczak '
+        write_command(filename, images, commandStart, jaszczak_sphere_diameters,
+                      cli_args)
+        plot_from_fom(filename, jaszczak_sphere_diameters, cli_args)
+
+    elif bool(cli_args['<nema7>']):
+        commandStart = 'target/release/foms -- nema7 '
```

Comment:
It turns out that the trailing space in `commandStart` is *crucial* for the correct functioning of the code.

This is bad. Let's try to remove this 'feature'.

Reply:
... and I suggest you do it like this:
```diff
- foms = get_foms(command + str(image_file_name))
+ foms = get_foms(f'{command} {image_file_name}')
```
in the definition of `write_command`.